### PR TITLE
Add Catalyst Handling

### PR DESCRIPTION
**Pull request by tvwenger**
Fixes https://github.com/lgfrbcsgo/du-factory-generator/issues/26

### CODE REVIEW

File: src/graph.ts [hunk 1]
```diff
@@ -477,7 +588,7 @@ export class FactoryGraph {
                 console.log(container)
                 throw new Error("Container exceeds outgoing link limit")
             }
-            if (container.egress > container.ingress) {
+            if (container.egress > 1.0001 * container.ingress) {
```
Comment (lgfrbcsgo) [inline src/graph.ts:591]: Do we run into issues with float point numbers here?

File: src/graph.ts [hunk 2]
```diff
@@ -462,6 +465,114 @@ export class FactoryGraph {
         return new Set(Array.from(this.containers).filter((node) => node.item === item))
     }
 
+    /**
+     * Handle the production and transfer of catalysts
+     * */
+    handleCatalysts() {
```
Comment (lgfrbcsgo) [inline src/graph.ts:471]: Not really an issue, but I would prefer this the be a standalone function which receives the factory as input. I would prefer the `FactoryGraph` to only contain as much logic as really necessary.

Comment (tvwenger) [reply src/graph.ts]: Fix in https://github.com/lgfrbcsgo/du-factory-generator/commit/5320bc0857710d3f4a1f2e6f9bd056f532d23d9b

File: src/graph.ts (hunk 1 above)
Comment (tvwenger) [reply src/graph.ts:488]: Ah yes, I forgot to mention this change. There is probably a more elegant way to handle this.

Comment (tvwenger) [reply src/graph.ts]: Fix in https://github.com/lgfrbcsgo/du-factory-generator/commit/5cc1a153e809752d2a07f246b0845d258a5da174

**Review by lgfrbcsgo — APPROVED** [review]

**Pull request merged**